conans/client/file_copier.py

```diff
@@ -238,6 +241,16 @@ def _copy_files(files, src, dst, keep_path, symlinks):
                     pass
                 os.symlink(linkto, abs_dst_name)  # @UndefinedVariable
             else:
-                shutil.copy2(abs_src_name, abs_dst_name)
+                inode = os.stat(abs_src_name).st_ino
+                if inode in inodes:
+                    try:
+                        os.link(abs_src_name, abs_dst_name)
+                    except OSError as error:
+                        # e.g. OSError: [Errno 18] Invalid cross-device link
+                        raise ConanException(
```

Comment:
I'd say it should fall back to the regular copy

Reply:
Regular copy won't prevent hard link, if we don't change the packaging flow, we won't be able to identify hard links before compressing files.

Reply:
I am not sure I understand. I mean doing something like:
```
try:
    os.link(abs_src_name, abs_dst_name)
except OSError:
    # file system doesn't support hard-links
    shutil.copy2(abs_src_name, abs_dst_name)
```
I think cross-file-system `os.link` may fail in many cases, especially if you're copying into network drive or flash drive.